java/datastructures/lib/src/main/java/datastructures/linkedlist/LinkedList.java

```diff
@@ -94,6 +94,26 @@ public boolean includes(int value) {
     return false;
   }
 
+  public int kthFromEnd(int k) {
+
+    if (head == null) {
+      throw new IllegalStateException("Linked List is empty");
+    }
+    Node slow = head;
+    Node fast = head;
+    for (int i = 0; i < k; i++) {
+      if (fast == null) {
```

Comment:
If `fast` is already null going into this if statement, the if statement will throw a null pointer exception and fail. You'll need to find a way to check whether `fast.next` is null and go from there (same thing for the while statement on line 110).